geode-core/src/main/java/org/apache/geode/internal/cache/PoolFactoryImpl.java

```diff
@@ -335,7 +335,7 @@ public Pool create(String name) throws CacheException {
         registry.creatingPool();
       }
     }
-    return PoolImpl.create(this.pm, name, this.attributes, this.locatorAddresses);
+    return PoolImpl.create(this.pm, name, this.attributes, this.locatorAddresses, cache);
```

Comment:
Note that "cache" here could be null. I didn't see the new PoolImpl code handling us passing it null.
The original PoolManager was designed so it could be started before the Cache or DistributedSystem. The new ClientCacheFactory by default creates it own pools which is the more normal way for people to create a pool and it is not a problem since those pools will know their Cache.
But the PoolManager is a problem since it lets us create pools before the cache exists. Later, it would look the singleton cache up using the getInstance methods. If you make this change I think the old code will just need to be broken. That is, you will not be able to create pool unless a Cache already exists. That is probably the direction we want to go but you might want to wait for geode 2.0 to make this breaking change